mycore-orcid2/src/main/resources/xsl/orcid2/v3/work2mcr_generic.xsl

```diff
@@ -0,0 +1,220 @@
+      </xsl:for-each>
+      <mods:title>
+        <xsl:value-of select="text()" />
+      </mods:title>
+    </mods:titleInfo>
+  </xsl:template>
+
+  <xsl:template match="work:journal-title">
+    <mods:titleInfo>
+      <mods:title>
+        <xsl:value-of select="text()"/>
+      </mods:title>
+    </mods:titleInfo>
+  </xsl:template>
+
+  <xsl:template match="work:contributors">
+    <xsl:apply-templates select="work:contributor" />
+  </xsl:template>
+
+  <xsl:template match="work:contributor">
+    <mods:name type="personal">
+      <xsl:apply-templates select="work:credit-name" />
+      <xsl:apply-templates select="work:contributor-attributes" />
+      <xsl:apply-templates select="common:contributor-orcid" />
+    </mods:name>
+  </xsl:template>
+  
+  <!-- Split "Doe, John F." into mods:namePart -->
+  <xsl:template match="work:credit-name[contains(.,',')]">
+    <mods:namePart type="family">
+      <xsl:value-of select="normalize-space(substring-before(.,','))" />
+    </mods:namePart>
+    <mods:namePart type="given">
+      <xsl:value-of select="normalize-space(substring-after(.,','))" />
+    </mods:namePart>
+  </xsl:template>
+  
+  <!-- Split "John F. Doe" into mods:namePart -->
+  <xsl:template match="work:credit-name[not(contains(.,','))]">
+    <mods:namePart type="family">
+      <xsl:value-of select="tokenize(.,' ')[last()]" />
+    </mods:namePart>
+    <mods:namePart type="given">
+      <xsl:for-each select="tokenize(.,' ')">
+        <xsl:if test="position() != last()">
+          <xsl:if test="position() &gt; 1">
+            <xsl:text> </xsl:text>
+          </xsl:if>
+          <xsl:value-of select="." />
+        </xsl:if>
+      </xsl:for-each>
+    </mods:namePart>
+  </xsl:template>
+
+  <xsl:template match="work:contributor-attributes">
+    <xsl:apply-templates select="work:contributor-role" />
+  </xsl:template>
+
+  <!-- ORCID.org contributor roles to MARC relator codes mapping -->
+  <xsl:template match="work:contributor-role">
+    <mods:role>
+      <mods:roleTerm type="code" authority="marcrelator">
+        <xsl:choose>
+          <xsl:when test=".='author'">aut</xsl:when>
+          <xsl:when test=".='assignee'">asg</xsl:when>
+          <xsl:when test=".='editor'">edt</xsl:when>
+          <xsl:when test=".='chair-or-translator'">trl</xsl:when>
+          <xsl:when test=".='co-investigator'">ctb</xsl:when>
+          <xsl:when test=".='co-inventor'">ctb</xsl:when>
+          <xsl:when test=".='graduate-student'">ctb</xsl:when>
+          <xsl:when test=".='other-inventor'">ctb</xsl:when>
+          <xsl:when test=".='principal-investigator'">rth</xsl:when>
+          <xsl:when test=".='postdoctoral-researcher'">res</xsl:when>
+          <xsl:when test=".='support-staff'">ctb</xsl:when>
+          <xsl:otherwise>aut</xsl:otherwise>
+        </xsl:choose>
+      </mods:roleTerm>
+    </mods:role>
+  </xsl:template>
+
+  <xsl:template match="common:contributor-orcid[@common:path]">
```

Comment:
the xpath should be 
`match="common:contributor-orcid[common:path]"` otherwise the orcid id of the contributor is not getting into the mods xml